test/AlarmClockTest.cpp

```diff
@@ -64,7 +65,7 @@ TEST_F(AlarmClockTest, microsecondsLessThan500ms) {
    WaitForAlarmClockToExpire(alerter);
    EXPECT_TRUE(alerter.Expired());
    EXPECT_GE(AlarmClockTest::mFakeSleepUs, us-kFakeSleepLeeway);
-   EXPECT_LE(AlarmClockTest::mFakeSleepUs, us);
+   EXPECT_LE(AlarmClockTest::mFakeSleepUs, static_cast<unsigned int>(us));
```

Comment:
Alright I did some digging on this one...

mFakeSleepUs takes an unsigned int because of a chain of function calls that ultimate lead to this comparison:

Line 79 of AlarmClock.h
```while (sw.ElapsedUs() < timeUsTillExpire) {```

We have to have a StopWatch object to keep the AlarmClock sleep precise, and the return value of ```sw.ElapsedUs()``` is an unsigned int. Therefore, we made the ```timeUsTillExpire``` function variable an unsigned int for this comparison.

I think it makes more sense to leave these static_casts in the tests, rather than consolidating them into one static cast and putting it in the actual AlarmClock class. I think changing the semantics of the class to make the tests slightly easier to read is a little backwards in my opinion. If anyone disagrees let me know. Otherwise, I will leave this part of the test as is.